### DISCUSSION TITLE
Is there a way to delete multiple records at once?

### DISCUSSION THREAD

**Discussion by 3ffe** [pocketbase/pocketbase#1108] [Q&A] [answered]
The SDK only has a method for deleting one record at a time, like this
```javascript
await pb.collection('demo').delete('YOUR_RECORD_ID');
```
But sometimes I would like to delete all records that match some condition in a collection. What is the best way to do this? Should I first fetch all the relevant records, get their ids, and then do a for loop to delete each of them?

Or could an interface for deleting a list of records be provided in the future?

Thanks.

**Comment by ganigeorgiev** [comment, marked as answer]
Currently this is not supported and as you noted, you'll have to first fetch the ids and then send individual delete requests.

Transactional batch create/update/delete is in my todo and will be implemented in the future in relation to #48, but currently the migrations flow and filters enhancements are with higher priority.

**Reply by 3ffe** [reply]
Thank you Gani!

**Comment by wolfsolver** [comment]
Just as reference (since I search for the same thing an reach this), is possible to truncate via admin console

**Reply by ganigeorgiev** [reply]
Please don't spam - https://github.com/pocketbase/pocketbase/discussions/2950#discussioncomment-16887888.
